### ISSUE TITLE
JupyterHub xeus-cling does not run

### ISSUE THREAD

**Issue by huntrods** [QuantStack/xeus-cling#221]
I am running Ubuntu 18.04 server. I installed JupyterHub (via anaconda) and all is working well. Some while ago I installed build-essential, then xeus-cling (via conda install).

It all worked well.

Recently I updated Ubuntu (apt-get update; apt-get upgrade) and also updated xeus (conda update -c conda-forge -c QuantStack xeus-cling). This was two days ago.

Ever since, I cannot start any C++ kernel in JupyterHub. All of them fail with the identical error in the JupyterHub log file:

Warning in cling::IncrementalParser::CheckABICompatibility():
  Possible C++ standard library mismatch, compiled with __GLIBCXX__ '20180125'
  Extraction of runtime standard library version was: '20180720'
[I 2019-02-10 21:58:24.886 SingleUserNotebookApp restarter:110] KernelRestarter: restarting kernel (4/5), new random ports
ERROR in cling::CIFactory::createCI(): cannot extract standard library include paths!
Invoking:
  LC_ALL=C x86_64-conda_cos6-linux-gnu-c++  -O3 -DNDEBUG -xc++ -E -v /dev/null 2>&1 >/dev/null | awk '/^#include </,/^End of search/{if (!/^#include </ && !/^End of search/){ print }}' | GREP_OPTIONS= grep -E "(c|g)\+\+"
Results was:
With exit code 256
In file included from input_line_1:1:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/c++/7.3.0/new:39:
In file included from /usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/x86_64-linux-gnu/c++/7.3.0/bits/c++config.h:533:
/usr/lib/gcc/x86_64-linux-gnu/7.3.0/../../../../include/x86_64-linux-gnu/c++/7.3.0/bits/os_defines.h:39:10: fatal error: 'features.h' file not found
#include <features.h>
         ^~~~~~~~~~~~
Warning in cling::IncrementalParser::CheckABICompatibility():
  Possible C++ standard library mismatch, compiled with __GLIBCXX__ '20180125'
  Extraction of runtime standard library version was: '20180720'
[W 2019-02-10 21:58:27.902 SingleUserNotebookApp restarter:100] KernelRestarter: restart failed

Is there a solution or work-around to this issue?
Thanks.

**Comment by SylvainCorlay** [comment]
Hey @huntrods, thanks for posting!

If you drop the `-c QuantStack`, that should just work. We finished the transition to conda-forge and don't recommend using the QuantStack channel anymore.

(The issue was a relocation issue of the gcc7 package on the QuantStack channel.)

**Comment by huntrods** [comment]
I tried that but am not having success.

I tried just updating xeus-cling without -c QuantStack, and that didn't work (it said all packages were already installed).

So I tried 'conda remove xeus-cling' followed by 'conda install -c conda-forge xeus-cling'. No errors, but it still won't run (same error in logs about missing <features.h>).

I tried shutting down JupyterHub and then repeating the steps above (remove & reinstall xeus-cling), but I'm getting the same error message involving a missing <features.h>.

**Comment by SylvainCorlay** [comment]
You would need to start from a fresh environment. 

If you do `conda list` you will see all installed packages and if any comes from QuantStack.

**Comment by huntrods** [comment]
Thanks very much. That worked, 

 I can't actually start from a fresh environment due to the way I built this JupyterHub, but I took your advice and ran "conda list | grep Quant" and found many packages still using QuantStack.

 I removed each one with "conda remove xxx" and then reinstalled xeus-cling from conda-forge, and it all  works perfectly now.

Just FYI, If you only remove xeus-cling, you are left with the following packages still coming from QuantStack: ‘xeus’, ‘clangdev’, ‘cling’, ‘cppzmq’, ‘cryptopp’, ‘gcc-7’, libgcc-7’, llvmdev’ . All must be removed before reinstalling xeus-cling from -c conda-forge.

Again, thanks very much.
